cli/src/pcluster/models/imagebuilder.py

```diff
@@ -144,39 +146,61 @@ def config(self):
             self.__config = ImageBuilderSchema().load(self.source_config)
         return self.__config
 
-    def create(self, disable_rollback: bool = False):
+    def create(
+        self,
+        disable_rollback: bool = True,
+        suppress_validators: bool = False,
+        validation_failure_level: FailureLevel = FailureLevel.ERROR,
+    ):
         """Create the CFN Stack and associate resources."""
         # validate image name
         self._validate_image_name()
 
-        # check imagebuilder stack existence
+        # check image existence
+        if AWSApi.instance().ec2.image_exists(self.image_name):
+            raise ImageBuilderActionError(f"ParallelCluster image {self.image_name} already exists")
+
+        # check stack existence
         if AWSApi.instance().cfn.stack_exists(self.image_name):
-            raise ImageBuilderActionError(f"ImageBuilder stack {self.image_name} already exists")
+            raise ImageBuilderActionError(
+                f"ParallelCluster build infrastructure for image {self.image_name} already exists"
+            )
 
-        validation_failures = self.config.validate()
-        if validation_failures:
-            # TODO skip validation errors
-            raise ImageBuilderActionError("Configuration is invalid", validation_failures=validation_failures)
+        if not suppress_validators:
+            validation_failures = self.config.validate()
+            for failure in validation_failures:
+                if failure.level.value > FailureLevel(validation_failure_level).value:
+                    # Raise the exception if there is a failure with a level greater than the specified one
+                    raise ImageBuilderActionError("Configuration is invalid", validation_failures=validation_failures)
 
         # Add tags information to the stack
-        tags = copy.deepcopy(self.config.build.tags) or []
-        tags.append(BaseTag(key="pcluster_build_image", value=get_installed_version()))
-        tags = [{"Key": tag.key, "Value": tag.value} for tag in tags]
+        cfn_tags = copy.deepcopy(self.config.build.tags) or []
+        cfn_tags.append(BaseTag(key="pcluster_build_image", value=get_installed_version()))
+        cfn_tags = [{"Key": tag.key, "Value": tag.value} for tag in cfn_tags]
```

Comment:
we can add a TODO to reminder adding cloudwatch tag and s3 bucket tag in cfn_tags.

Reply:
done